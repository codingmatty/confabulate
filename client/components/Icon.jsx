import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledIcon = styled.i`
  font-size: ${({ size }) => size};
  font-weight: ${({ bold }) => bold && 'bold'};
`;

export default function Icon({ type, bold, size }) {
  return (
    <StyledIcon className="material-icons" size={size} bold={bold}>
      {type}
    </StyledIcon>
  );
}
Icon.propTypes = {
  type: PropTypes.string,
  bold: PropTypes.bool,
  size: PropTypes.string
};
