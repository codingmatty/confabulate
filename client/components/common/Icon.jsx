import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledIcon = styled.i`
  font-size: inherit;
  font-weight: ${({ bold }) => bold && 'bold'};
`;

export default function Icon({ type, bold, className = '' }) {
  return (
    <StyledIcon className={`material-icons ${className}`} bold={bold}>
      {type}
    </StyledIcon>
  );
}
Icon.propTypes = {
  bold: PropTypes.bool,
  className: PropTypes.string,
  type: PropTypes.string
};
