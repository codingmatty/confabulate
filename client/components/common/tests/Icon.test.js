import { cleanup, render } from 'react-testing-library';
import Icon from '../Icon';

describe('Icon', () => {
  afterEach(cleanup);

  it('matches default snapshot', () => {
    const { container } = render(<Icon type="test" />);
    expect(container).toMatchSnapshot();
  });

  it('sets className', () => {
    const { container } = render(<Icon type="test" className="test-class" />);
    expect(container.firstChild).toHaveClass('test-class');
  });

  it('adds bold font-weight', () => {
    const { container } = render(<Icon type="test" bold />);
    expect(container.firstChild).toHaveStyleRule('font-weight', 'bold');
  });
});
