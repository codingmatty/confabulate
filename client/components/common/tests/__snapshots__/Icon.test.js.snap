// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon matches default snapshot 1`] = `
.c0 {
  font-size: inherit;
}

<div>
  <i
    class="material-icons  c0"
  >
    test
  </i>
</div>
`;
